soh/soh/Enhancements/randomizer/3drando/hints.cpp

```diff
@@ -1074,8 +1075,9 @@ void CreateStoneHints() {
 
   while(totalStones != 0){
```

Comment:
Doesn't the change directly below this line make this redundant now? Cleaning up this second to have distribute first in the loop would better be a larger change, something like this, replacing lines 1074 through 1082. 

```suggestion
    while(totalStones != 0){
   DistributeHints(selectedHints, totalStones, distTable, hintSetting.junkWeight);
   totalStones = PlaceHints(selectedHints, distTable);
  }
```

Reply:
I think the main issue is that we have to attempt one cycle in order for totalStones to have a value to check for the whiel loop. That being said this does sound like the perfect case for a do-while loop.

Reply:
Oh nevermind, forgot about the initial `totalStones` value coming from `GetEmptyGossipStones`.

Reply:
There is one thing that's different about the `DistributeHints` call inside vs outside the loop, and that's the extra `false` at the end. That `false` prevents re-adding any fixed hints that were added in the first call to `DistributeHints`.